docs/develop/java/message-passing.mdx

```diff
@@ -33,16 +33,20 @@ Temporal Clients use messages to read Workflow state and control execution.
 See [Workflow message passing](/encyclopedia/workflow-message-passing) for a general overview of this topic.
 This page introduces these features for the Temporal Java SDK.
 
+## Write message handlers {#writing-message-handlers}
+
 :::info
-The code that follows is part of a working message passing (URL TODO).
+The code that follows is part of a working message passing <div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**URL TODO**</div>.
 :::
 
-General guidelines for writing message handlers:
+Follow these guidelines when writing your message handlers:
 
-- Arguments and return values of handlers (and the main Workflow function) must be [serializable](/dataconversion): a class will often be the right choice.
-- It's possible to write handler methods that take multiple arguments, but this is not recommended: instead use a single class argument to which fields can be added as needed.
+- Message handlers are defined as methods on the Workflow class, using one of the three decorators: [`@workflow.query`](https://python.temporal.io/temporalio.workflow.html#query), [`@workflow.signal`](https://python.temporal.io/temporalio.workflow.html#signal), and [`@workflow.update`](https://python.temporal.io/temporalio.workflow.html#update).
+- The parameters and return values of handlers and the main Workflow function must be [serializable](/dataconversion).
+- Prefer [data classes](https://docs.python.org/3/library/dataclasses.html) to multiple input parameters.
+  Data class parameters allow you to add fields without changing the calling signature.
```

Comment:
Just leaving a TODO here for conversion to Java

Reply:
D'erp again. Can't believe I missed this. Fixing.